node/bin/src/batch_verification_manager.rs

```diff
@@ -0,0 +1,265 @@
+    Ok(())
+}
+
+pub struct BatchVerifier {
+    config: BatchVerificationConfig,
+    request_id_counter: AtomicU64,
+    server: Arc<BatchVerificationServer>,
+    response_channels: Arc<DashMap<u64, mpsc::Sender<BatchVerificationResponse>>>,
+}
+
+#[derive(Debug, thiserror::Error)]
+enum BatchVerificationError {
+    #[error("Timeout")]
+    Timeout,
+    #[error("Not enough signers")]
+    NotEnoughSigners,
+    #[error("Internal error: {0}")]
+    Internal(String),
+}
+
+impl From<BatchVerificationRequestError> for BatchVerificationError {
+    fn from(err: BatchVerificationRequestError) -> Self {
+        match err {
+            BatchVerificationRequestError::NotEnoughClients => {
+                BatchVerificationError::NotEnoughSigners
+            }
+        }
+    }
+}
+
+impl BatchVerificationError {
+    fn retryable(&self) -> bool {
+        !matches!(self, BatchVerificationError::Internal(_))
+    }
+}
+
+impl BatchVerifier {
+    pub fn new(
+        config: BatchVerificationConfig,
+        response_channels: Arc<DashMap<u64, mpsc::Sender<BatchVerificationResponse>>>,
+        server: Arc<BatchVerificationServer>,
+    ) -> Self {
+        Self {
+            config,
+            request_id_counter: AtomicU64::new(1),
+            response_channels,
+            server,
+        }
+    }
+
+    async fn run<E: Send + Sync>(
+        &self,
+        mut batch_for_signing_receiver: Receiver<BatchForSigning<E>>,
+        singed_batcher_sender: Sender<SignedBatchEnvelope<E>>,
+    ) -> anyhow::Result<()> {
+        loop {
+            let Some(batch_envelope) = batch_for_signing_receiver.recv().await else {
+                // Channel closed, exit the loop
+                tracing::info!("BatchForSigning channel closed, exiting verifier",);
+                break Ok(());
+            };
+            let mut retry_count = 0;
+            let deadline = Instant::now() + self.config.total_timeout;
+            let signatures = loop {
+                match self.verify_batch(&batch_envelope).await {
+                    Ok(result) => break Ok(result),
+                    Err(err) if err.retryable() => {
+                        if Instant::now() < deadline {
+                            retry_count += 1;
+                            tracing::warn!(
+                                "Batch verification failed, attempt {} retrying. Error: {}",
+                                retry_count,
+                                err
+                            );
+
+                            tokio::time::sleep(self.config.retry_delay).await;
+                        } else {
+                            tracing::warn!(
+                                "Batch verification failed after {} retries exceeding total timeout. Bailing. Last error: {}",
+                                retry_count,
+                                err
+                            );
+                            break Err(err);
```

Comment:
I think this is inline with other cases when we are currently crashing the server, but maybe crashes should be limitted to truly unrecoverable errors. Here it could make sense to keep trying infinitely to verify the batch and wait for an engineer to debug the problem.